### PR TITLE
Feature/11&13 task/vi/11&13 task

### PR DESCRIPTION
**Pull request by VsIakushev**
Экраны "Профиль" и "Данные пользователя"

https://github.com/dimapaull/UIKitTasks/assets/114725077/a55c6441-a3a3-4ebb-a702-19b823ab4d31

Снимок экрана 2024-02-15 в 10 11 16 PM

### CODE REVIEW

File: UIKitTasks/UIKitTasks/Controllers/UserDataViewController.swift [hunk 1]
```diff
@@ -3,10 +3,493 @@
 
 import UIKit
 
-/// Контроллер предоставляющий пользовательские данные
+@propertyWrapper
+struct UserDefault<T> {
```
[the remark this answers is not in the document]
Comment (VsIakushev) [reply UIKitTasks/UIKitTasks/Controllers/UserDataViewController.swift:7]: done

File: UIKitTasks/UIKitTasks/Controllers/ProfileViewController.swift [hunk 2]
```diff
@@ -5,8 +5,328 @@ import UIKit
...
+        separator.translatesAutoresizingMaskIntoConstraints = false
+        separator.backgroundColor = Constants.separatorColor
+        separator.widthAnchor.constraint(equalToConstant: 335).isActive = true
+        separator.heightAnchor.constraint(equalToConstant: 1).isActive = true
+        separator.leftAnchor.constraint(equalTo: feedbackButton.leftAnchor).isActive = true
+        separator.bottomAnchor.constraint(equalTo: feedbackButton.bottomAnchor, constant: -1).isActive = true
+    }
+
+    @objc private func qrCodeButtonPressed() {
+        print("QR Pressed")
```
[the remark this answers is not in the document]
Comment (VsIakushev) [reply UIKitTasks/UIKitTasks/Controllers/ProfileViewController.swift:310]: deleted

[the remark this answers is not in the document]
Comment (VsIakushev) [reply UIKitTasks/UIKitTasks/Controllers/ProfileViewController.swift:310]: deleted

File: UIKitTasks/UIKitTasks/Controllers/ProfileViewController.swift [hunk 3]
```diff
@@ -5,8 +5,328 @@ import UIKit
...
+    }
+
+    @objc private func cardInfoButtonPressed() {
+        print("Card Info Pressed")
```
[the remark this answers is not in the document]
Comment (VsIakushev) [reply UIKitTasks/UIKitTasks/Controllers/ProfileViewController.swift:314]: deleted

File: UIKitTasks/UIKitTasks/Controllers/ProfileViewController.swift [hunk 4]
```diff
@@ -5,8 +5,328 @@ import UIKit
...
+    @objc private func userInfoButtonPressed() {
+        print("User Info Pressed")
+        let userDataVC = UserDataViewController()
+        let navigationController = UINavigationController(rootViewController: userDataVC)
+        navigationController.modalPresentationStyle = .fullScreen
+        present(navigationController, animated: true)
+    }
+
+    @objc private func inviteButtonPressed() {
+        print("Invite Pressed")
```
[the remark this answers is not in the document]
Comment (VsIakushev) [reply UIKitTasks/UIKitTasks/Controllers/ProfileViewController.swift:326]: deleted

File: UIKitTasks/UIKitTasks/Controllers/SizePickerViewController.swift [hunk 5]
```diff
@@ -0,0 +1,152 @@
+// SizePickerViewController.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+/// Всплывающее Вью с выбором размера обуви
+class SizePickerViewController: UIViewController {
```
[the remark this answers is not in the document]
Comment (VsIakushev) [reply UIKitTasks/UIKitTasks/Controllers/SizePickerViewController.swift:7]: done

File: UIKitTasks/UIKitTasks/Controllers/UserDataViewController.swift [hunk 6]
```diff
@@ -3,10 +3,493 @@
...
+            lastName = textField.text ?? ""
+        } else if textField == phoneTextField {
+            phoneNumber = textField.text ?? ""
+        } else if textField == shoeSizeTextField {
+            shoeSize = textField.text ?? ""
+        } else if textField == dateOfBirthTextField {
+            dateOfBirth = textField.text ?? ""
+        } else if textField == emailTextField {
+            email = textField.text ?? ""
+        }
```
[the remark this answers is not in the document]
Comment (VsIakushev) [reply UIKitTasks/UIKitTasks/Controllers/UserDataViewController.swift:487]: done

**Review by dimapaull — APPROVED** [review]

**Pull request merged**